lib/ddtrace/tracer.rb

```diff
@@ -21,12 +22,26 @@ class Tracer
     # namespace. This logger outputs to +STDOUT+ by default, and is considered thread-safe.
     def self.log
       unless defined? @logger
-        @logger = Logger.new(STDOUT)
+        @logger = Datadog::Logger.new(STDOUT)
         @logger.level = Logger::WARN
       end
       @logger
     end
 
+    # Override the default logger with a custom one.
+    def self.log=(logger)
+      return nil unless logger
+      return nil unless logger.respond_to? :methods
+      return nil unless logger.respond_to? :error
+      if logger.respond_to? :methods
+        unimplemented = Logger.new(STDOUT).methods - logger.methods
+        unless unimplemented.empty?
+          logger.error("logger #{logger} does not implement #{unimplemented}")
```

Comment:
do we need to `return` here? I mean if the new logger doesn't implement the `Logger` "interface", is it right to set it anyway?